alacritty/src/input.rs

```diff
@@ -231,6 +231,41 @@ impl<T: EventListener> Execute<T> for Action {
             Action::Search(SearchAction::SearchDeleteWord) => ctx.search_pop_word(),
             Action::Search(SearchAction::SearchHistoryPrevious) => ctx.search_history_previous(),
             Action::Search(SearchAction::SearchHistoryNext) => ctx.search_history_next(),
+            Action::Mouse(MouseAction::ExpandSelection) => {
+                let selection_type = match ctx.mouse().click_state {
+                    ClickState::Click => {
+                        if ctx.modifiers().ctrl() {
+                            Some(SelectionType::Block)
+                        } else {
+                            Some(SelectionType::Simple)
+                        }
+                    },
+                    ClickState::DoubleClick => Some(SelectionType::Semantic),
+                    ClickState::TripleClick => Some(SelectionType::Lines),
+                    ClickState::None => None,
+                };
+
+                if let Some(selection_type) = selection_type {
+                    // Load mouse point, treating message bar and padding as the closest cell.
+                    let display_offset = ctx.terminal().grid().display_offset();
+                    let point = ctx.mouse().point(&ctx.size_info(), display_offset);
+
+                    let cell_side = ctx.mouse().cell_side;
+
+                    let selection = match &mut ctx.terminal_mut().selection {
+                        Some(selection) => selection,
+                        None => return,
+                    };
+
+                    selection.ty = selection_type;
+                    ctx.update_selection(point, cell_side);
+
+                    // Move vi mode cursor to mouse click position.
+                    if ctx.terminal().mode().contains(TermMode::VI) && !ctx.search_active() {
+                        ctx.terminal_mut().vi_mode_cursor.point = point;
+                    }
```

Comment:
Should this remain in `expand_selection` or perhaps be in `on_right_click` or handled automatically by `update_selection`?

Reply:
The vi mode cursor shouldn't be moved on right click unless the action is bound.